distributed/deploy/spec.py

```diff
@@ -333,7 +333,6 @@ def _correct_state(self):
                 await asyncio.wait(workers)
```

Comment:
I think you want to swap this for `gather` to ensure that all exceptions are retrieved (stops warning) and raised (should fix tests).

```suggestion
                 await asyncio.gather(*workers)
```

Reply:
Good point. Thanks Jim! Let's give it a try 🙂